OpenSim/Simulation/SimbodyEngine/Joint.h

```diff
@@ -131,30 +113,65 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Joint, ModelComponent);
                             direction of the Joint in the multibody tree. 
-                            Default is false (that is, forward).
+                            Default is false (that is, forward. parent to child).
         */
-    Joint(const std::string &name,
-          const PhysicalFrame& parent,
-          const SimTK::Vec3& locationInParent,
-          const SimTK::Vec3& orientationInParent,
-          const PhysicalFrame& child,
-          const SimTK::Vec3& locationInChild,
-          const SimTK::Vec3& orientationInChild,
-          bool reverse = false);
+    Joint( const std::string& name,
+           const std::string& parentName,
+           const std::string& childName,
+           bool reverse = false);
+
+    /** DEPRECATED Convenience Constructor
+    Create a Joint where the parent and child are specified as well as the
+    joint frames in the child and parent bodies in terms of their location
+    and orientation in their respective physical frames. Also an advanced option
+    to specify the tree structure to be constructed in the reverse direction,
+    that is child to parent, but the coordinates remain as if defined parent
+    to child. This can be useful for defining models from the ground up, yet
+    maintaining the convention of the knee, for example, of the relative
+    motion of the tibia (child) w.r.t. the femur (parent).
+
+    @param[in] name     the name associated with this joint (should be
+                        unique from other joints in the same model)
+    @param[in] parent   the parent PhysicalFrame that joint connects to
+    @param[in] locationInParent    Vec3 of the location of the joint in the
+                                   parent frame.
+    @param[in] orientationInParent Vec3 of the XYZ body-fixed Euler angles
+                                   of the joint frame orientation in the
+                                   parent frame.
+    @param[in] child    the child PhysicalFrame that joint connects to
+    @param[in] locationInChild     Vec3 of the location of the joint in the
+                                   child physical frame.
+    @param[in] orientationInChild  Vec3 of the XYZ body-fixed Euler angles
+                                   of the joint frame orientation in the
+                                   child physical frame.
+    @param[in] reverse  Advanced optional flag (bool) specifying the
+                        direction of the Joint in the multibody tree.
+                        Default is false (that is, forward).
+    */
+    DEPRECATED_14("Use Joint(name, parentName, childName) instead.")
+    Joint(const std::string& name,
+        const PhysicalFrame& parent,
+        const SimTK::Vec3& locationInParent,
+        const SimTK::Vec3& orientationInParent,
+        const PhysicalFrame& child,
+        const SimTK::Vec3& locationInChild,
+        const SimTK::Vec3& orientationInChild,
+        bool reverse);
```

Comment:
I think these deprecated constructors should take `Body`, not `PhysicalFrame`. That's what the old constructors took. It sounds messy for people to be specifying their own custom PhysicalOffsetFrames and then applying *additional* offsets via the Vec3's.